oneanddone/tests/functional/tests/test_task_assignment.py

```diff
@@ -0,0 +1,41 @@
+# This Source Code Form is subjectfrom django import forms to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.home import HomePage
+
+
+class TestAvailableTasks():
+    @pytest.mark.nondestructive
+    def test_assign_tasks(self, base_url, selenium, nonrepeatable_task, new_user):
+        home_page = HomePage(selenium, base_url).open()
+        assert not home_page.is_user_logged_in
+
+        edit_profile = home_page.login(new_user)
+        assert edit_profile.is_user_logged_in
+
+        available_tasks_page = home_page.click_available_tasks()
+        assert available_tasks_page.is_available_tasks_list_visible
+        assert len(available_tasks_page.available_tasks) > 0
+
+        nonrepeatable_task = available_tasks_page.available_tasks[0]
+        task_name = nonrepeatable_task.name
+        task_details = nonrepeatable_task.click()
+        assert task_name == task_details.name
+        assert task_details.is_get_started_button_visible
+        task_details.click_get_started_button()
+        assert task_details.is_abandon_task_button_visible
+        assert task_details.is_complete_task_button_visible
+        assert task_details.is_save_for_later_button_visible
+
+    @pytest.mark.nondestructive
+    def test_reassign_task(self, base_url, selenium, nonrepeatable_task, new_user):
+        home_page = HomePage(selenium, base_url).open()
+        assert not home_page.is_user_logged_in
+        edit_profile = home_page.login(new_user)
+        assert edit_profile.is_user_logged_in
+        available_tasks_page = home_page.click_available_tasks()
+        home_page.search_for_task(nonrepeatable_task.name)
+        assert len(available_tasks_page.available_tasks) == 0
```

Comment:
Only tasks that can be assigned to that particular user will appear as results for the search. Because that task has already been assigned in the previous test, I am checking if the task is listed.